### PR TITLE
Iss1144: Neo4j Add Tests for searchbyGeneId()

### PR DESCRIPTION
**Pull request by lindajiawenli**
Now that addDocumentToNeo4j() has been made and tests have been written for it, it is possible to add a large number of nodes/edges quickly.

A test suite to see if seachByGeneId( id ) still performs as desired is in order.
This function takes in an id (a String in the form of dbName:dbId, like 'ncbigene:207') and returns null or an object with 2 fields:

an array of relationships/edges/interactions
an array of neighbouring nodes

### Proposal

Precondition: Add all 5 Ben Goult documents to Neo4j (https://github.com/PathwayCommons/factoid/issues/1141)

 
- [x] Search for KANK1 (connected to 3 neighbouring nodes, 3 interactions)
- [x]  Seach for TLN1 (connected to 5 *unique* neighbouring nodes, 6 interactions)
- [x]  Search for TLNRD1 (connected to 5 neighbouring nodes (including itself), 5 interactions)
- [x]  Search for MAPK6 (should not exist)

### CODE REVIEW

**Comment by maxkfranz** [comment]
Looks pretty good.

Is your precondition handled manually for now?  If not, would it be sensible to load in the data necessary for your tests in a setup function?

File: src/neo4j/neo4j-functions.js [hunk 1]
```diff
@@ -110,10 +111,10 @@ export async function getNeighbouringNodes(id) {
  */
 export async function getInteractions(id) {
     let record = await searchByGeneId(id);
```
[the remark this answers is not in the document]
Comment (lindajiawenli) [reply src/neo4j/neo4j-functions.js:113]: Ah, you're right- we did discuss this on Wednesday but it slipped my mind.

I will do:

1. Rename the nodes from `Gene` to `Molecule`
2. Rename the `searchByGeneId` to `searchByNodeId`
3. Clean up lint errors 

And re-request your review!

**Comment by lindajiawenli** [comment]
> Is your precondition handled manually for now? If not, would it be sensible to load in the data necessary for your tests in a setup function?

Hi @maxkfranz, my precondition is currently in a before() block in my `search-tests.js` test suite. I thought about putting it in a beforeEach() block, but my tests only involve reading the same data so I didn't see a point.

Do you think this is good or is there a better way I can do it?

**Review by jvwong — APPROVED** [review]
`entity` is probably more in line with Biofactoid, but we all get the idea.

**Pull request merged**